AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Migration/MutationSyncMetadataMigrationDelegate+SQLite.swift

```diff
@@ -0,0 +1,123 @@
+        switch step {
+        case .emptyMutationSyncMetadataStore:
+            try emptyMutationSyncMetadataStore()
+        case .emptyModelSyncMetadataStore:
+            try emptyMutationSyncMetadataStore()
+        case .removeMutationSyncMetadataCopyStore:
+            try removeMutationSyncMetadataCopyStore()
+        case .createMutationSyncMetadataCopyStore:
+            try createMutationSyncMetadataCopyStore()
+        case .backfillMutationSyncMetadata:
+            try backfillMutationSyncMetadata()
+        case .removeMutationSyncMetadataStore:
+            try removeMutationSyncMetadataStore()
+        case .renameMutationSyncMetadataCopy:
+            try renameMutationSyncMetadataCopy()
+        }
+    }
+
+    // MARK: - Clear
+
+    @discardableResult func emptyMutationSyncMetadataStore() throws -> String {
+        guard let storageAdapter = storageAdapter else {
+            log.debug("Missing SQLiteStorageEngineAdapter")
+            throw DataStoreError.unknown("Missing storage adapter for model migration", "", nil)
+        }
+
+        return try storageAdapter.emptyStore(for: MutationSyncMetadata.schema)
+    }
+
+    @discardableResult func emptyModelSyncMetadataStore() throws -> String {
+        guard let storageAdapter = storageAdapter else {
+            log.debug("Missing SQLiteStorageEngineAdapter")
+            throw DataStoreError.unknown("Missing storage adapter for model migration", "", nil)
+        }
+
+        return try storageAdapter.emptyStore(for: ModelSyncMetadata.schema)
+    }
+
+    // MARK: - Migration
+
+    @discardableResult func removeMutationSyncMetadataCopyStore() throws -> String {
+        guard let storageAdapter = storageAdapter else {
+            log.debug("Missing SQLiteStorageEngineAdapter")
+            throw DataStoreError.unknown("Missing storage adapter for model migration", "", nil)
+        }
+
+        return try storageAdapter.removeStore(for: MutationSyncMetadataMigration.MutationSyncMetadataCopy.schema)
+    }
+
+    @discardableResult func createMutationSyncMetadataCopyStore() throws -> String {
+        guard let storageAdapter = storageAdapter else {
+            log.debug("Missing SQLiteStorageEngineAdapter")
+            throw DataStoreError.unknown("Missing storage adapter for model migration", "", nil)
+        }
+
+        return try storageAdapter.createStore(for: MutationSyncMetadataMigration.MutationSyncMetadataCopy.schema)
+    }
+
+    @discardableResult func backfillMutationSyncMetadata() throws -> String {
+        var sql = ""
+        for modelSchema in modelSchemas {
+            let modelName = modelSchema.name
+
+            if sql != "" {
+                sql += " UNION ALL "
+            }
+            sql += "SELECT id, \'\(modelName)\' as tableName FROM \(modelName)"
+        }
+        sql = "INSERT INTO \(MutationSyncMetadataMigration.MutationSyncMetadataCopy.modelName) (id,deleted,lastChangedAt,version) " +
```

Comment:
nit: can we perhaps get the list of field names from the schema here?

Reply:
from the initial glance, there is more than one place that uses the fields as a hard coded string: `(id,deleted,lastChangedAt,version)` and `select models.tableName || '|' || mm.id, mm.deleted, mm.lastChangedAt, mm.version`  so even if we got the fields from the schema in the first spot, we're then hardcoding them again in the next line, so we should it for both. Seems like it will get more obfuscated by doing so